js/queen.js

```diff
@@ -70,7 +83,11 @@ export class Queen extends Piece{
          * Get col decreasing moves
          */
         for (let i = this.colPos - 1; i >= minCols; i--) {
-            if (board[this.rowPos][i] !== undefined && board[this.rowPos][i].getPlayerId() == this.playerId) {
+            if (board[this.rowPos][i] && board[this.rowPos][i].getPlayerId() == this.playerId) {
```

Comment:
`===` should be used instead of `==` on line `86`.

Reply:
Updated.